### PR TITLE
[Docs] Add more info about running tests on iOS

### PR DESCRIPTION
**Pull request by bartekpacia**
I wish this info was there, that'd save me from looking inside the script. 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by chinmaygarde** [comment]
cc @jmagman

File: testing/ios/IosUnitTests/README.md [hunk 1]
```diff
@@ -1,16 +1,39 @@
...
 flutter/testing/run_tests.py --type=objc
 ```
 
+And if you're on Apple Silicon:
+
+```sh
+./flutter/testing/run_tests.py \
+  --type=objc \
+  --variant=host_debug_unopt_arm64 \
+  --ios=ios_debug_sim_unopt_arm64
```
Comment (jmagman) [inline testing/ios/IosUnitTests/README.md:20]: Shouldn't this be `--ios-variant`?
```suggestion
  --ios-variant ios_debug_sim_unopt_arm64
```

File: testing/ios/IosUnitTests/README.md [hunk 2]
```diff
@@ -1,16 +1,39 @@
...
+```
+
+The `.xcresult` is automatically removed after testing ends. To change this:
+
+```sh
+export FLUTTER_TEST_OUTPUTS_DIR=~/Desktop
```
Comment (jmagman) [inline testing/ios/IosUnitTests/README.md:26]: We set `FLUTTER_TEST_OUTPUTS_DIR` env in our CI system, however if it's useful to be used on the command line `parser.add_argument` output option around here
https://github.com/flutter/engine/blob/9b716e031fd2160213534add72489f18ca4b8361/testing/run_tests.py#L1219
And use that and fallback to `FLUTTER_TEST_OUTPUTS_DIR` is currently used, first  if it's not passed in.

https://github.com/flutter/engine/blob/9b716e031fd2160213534add72489f18ca4b8361/testing/run_tests.py#L831
https://github.com/flutter/engine/blob/9b716e031fd2160213534add72489f18ca4b8361/testing/run_tests.py#L297

However if you don't want to touch the python code I understand. 🙂

File: testing/ios/IosUnitTests/README.md (hunk 1 above)
Comment (bartekpacia) [reply testing/ios/IosUnitTests/README.md:20]: That's right! But there's no difference in behavior between:

```
./flutter/testing/run_tests.py \
  --type=objc \
  --variant=host_debug_unopt_arm64 \
  --ios=ios_debug_sim_unopt_arm64
```

and

```
$ ./flutter/testing/run_tests.py \
  --type=objc \
  --variant=host_debug_unopt_arm64 \
  --ios-variant=ios_debug_sim_unopt_arm64
```

Apparently Python's argparser sees that the only option that starts with `ios` is `ios-variant` and it all goes well thanks to this.

Anyway thanks for the suggestion, fixed.

File: testing/ios/IosUnitTests/README.md (hunk 2 above)
Comment (bartekpacia) [reply testing/ios/IosUnitTests/README.md:25]: For now I'd prefer to leave it as it is (I'm low on time), but I might come back to it in the future and add a proper argument for that.

**Review by jmagman — APPROVED** [review]
LGTM

**Comment by jmagman** [comment]
@vashworth can you give it a second review?

**Review by vashworth — APPROVED** [review]
LGTM! Thanks for adding this!

**Comment by bartekpacia** [comment]
Thanks!

**Pull request merged**